### ISSUE TITLE
Subdomains pointing to vercel.com are vulnerable

### ISSUE THREAD

**Issue by ScrubsAndStats** [EdOverflow/can-i-take-over-xyz#183]
## Service name
Vercel
## Proof
Successful subdomain takeover on a harvard.edu subdomain (screenshot).
![proof-vercel](https://user-images.githubusercontent.com/40786065/101316778-f7d68d00-387e-11eb-8078-258558cb5d56.png)

## Documentation
- Create a new repository on Github and upload an index.html
- Visit https://vercel.com/ and sign up using your Github account
- Create a new project and point it to the previously created Github repository
- Open the "Domains" tab on Vercel and add the vulnerable domain
- Boom! Exploited!

**Comment by marcelo321** [comment]
Can you share the cname regex and the fingerprint?

**Comment by c0d3d-v3n0m** [comment]
> 
> 
> Can you share the cname regex and the fingerprint?

Sure

`{
			"service": "vercel",
			"cname": [
					""
			],
			"fingerprint": [
					"The deployment could not be found on Vercel."
			],
			"nxdomain": false
	}`

**Comment by adityathebe** [comment]
There are definitely edge cases here.

```bash
$ host -t CNAME anythingrandom.console.dev.twilio.com
anythingrandom.console.dev.twilio.com is an alias for cname.vercel-dns.com.
```

```bash
$ curl 'https://anythingrandom.console.dev.twilio.com/'                                                                                                     10:12:48
The deployment could not be found on Vercel.

DEPLOYMENT_NOT_FOUND
```

![image](https://user-images.githubusercontent.com/13419481/103145470-65c9f200-4763-11eb-95fe-65827e41b12a.png)

**Comment by marcelo321** [comment]
so the cname we need to grep is vercel-dns.com not vercel.com. thank you @adityathebe

**Comment by blackcodersec** [comment]
> > Can you share the cname regex and the fingerprint?
> 
> Sure
> 
> `{ "service": "vercel", "cname": [ "" ], "fingerprint": [ "The deployment could not be found on Vercel." ], "nxdomain": false }`

are you takeover any subdomain? Do you have any poc?

**Comment by raladev** [comment]
Summary for 2021:
U can takeover mashed.potato.com only if potato.com is not used in the account of the victim, otherwise, u will get Already owned err.

**Comment by jan-muhammad-zaidi** [comment]
This can be closed as Edge-case

**Comment by M359AH** [comment]
It still vulnerable yesterday I takeover 2 subdomains and I've upload my index

**Comment by jan-muhammad-zaidi** [comment]
@M359AH u took over mashed.potato.com  even when potato.com is already registered? If yes, please share how you managed to do that? Just curious :0

**Comment by M359AH** [comment]
@jan-muhammad-zaidi 

Hello Muhammed 

I've found the subdomain I got this error page

![Screen Shot 2022-02-07 at 12 20 49 PM](https://user-images.githubusercontent.com/57077443/152773723-7c3c5a56-4d9d-4e4e-a0b6-ce6c9404fdb6.jpg)

* After it, I go to see the CNAME 

<img width="880" alt="Screen Shot 2022-02-07 at 12 42 20 PM" src="https://user-images.githubusercontent.com/57077443/152773815-6b8149b8-451a-47dd-bc99-b4d8acd1e385.png">

Now I go to vercel.app and add a public repository contains my PoC index and after import the project I've add the domain and added successfully 

and my PoC has been uploaded

**Comment by jan-muhammad-zaidi** [comment]
How come it's not showing a domain already registered error? Like this
![image](https://user-images.githubusercontent.com/58605984/152777479-39cada26-5e47-46b2-a3af-96eceb908de8.png)

**Comment by M359AH** [comment]
Hello @jan-muhammad-zaidi 

I think your target is not vulnerable because It should be registered without an errors like my comment above

**Comment by M359AH** [comment]
Your index should be uploaded like It:

<img width="1440" alt="Screen Shot 2022-02-07 at 3 38 12 PM" src="https://user-images.githubusercontent.com/57077443/152798869-0c606792-e8df-4309-a6c5-cdc9a1b2811d.png">


Sorry for my bad image edit 😅 😂 😂

**Comment by jan-muhammad-zaidi** [comment]
@M359AH no issues with the edit though :P

**Comment by umar98** [comment]
![Untitled](https://user-images.githubusercontent.com/42386653/169685319-16d81ad3-48ef-4322-97e7-35ceb7f8d0ba.png)


Any luck on how to do this?

**Comment by Fatmanpoc** [comment]
> ![Untitled](https://user-images.githubusercontent.com/42386653/169685319-16d81ad3-48ef-4322-97e7-35ceb7f8d0ba.png)
> 
> Any luck on how to do this?

got same error...any clue on this?

**Comment by jareddarkweb** [comment]
me aswelll

**Comment by Faizee-Asad** [comment]
https://vercel.com/docs/concepts/projects/custom-domains

**Comment by tw-ddis** [comment]
> ![Untitled](https://user-images.githubusercontent.com/42386653/169685319-16d81ad3-48ef-4322-97e7-35ceb7f8d0ba.png)
> 
> Any luck on how to do this?

same error , vercel fixed the bug no luck

**Comment by abuvanth** [comment]
no more takeover

**Comment by joren485** [comment]
Domain takeovers using Vercel are definitely still possible.

However, they are limited. In my testing, I found that a domain is not vulnerable if:
* The root domain is used by a Vercel account (i.e. the root domain points to `76.76.21.21` and is linked to a project).
* The domain/root domain is verified, even if the root domain does not point to `76.76.21.21`.
* Another subdomain of the same root domain is used by a Vercel account.

In practice, this means many subdomains will not be vulnerable (but subdomains definitely can be vulnerable).

There seems to be only one way to be sure a domain is vulnerable or not: try it out.

I created a PR to update the README: #375

**Comment by 0xmahi** [comment]
![image](https://github.com/EdOverflow/can-i-take-over-xyz/assets/66868138/38e8f292-b4d2-4c1d-8d1f-f24cebcfe304)

**Comment by aravindb26** [comment]
I have the same error but it can be only possible if we configure DNS but to add it the domain should be shown in Domains category the how could we add DNS?

**Comment by excommunicado110** [comment]
Any success on this?

**Comment by brijesh1353** [comment]
I have the same error but it can be only possible if we configure DNS to that custom domain that should be shown in the Domains category but it's not showing, how could we add DNS?

This has happened to me too, please show me the solution

**Comment by xElkomy** [comment]
It's not possible anymore because you have to add a txt record, and that is not possible in the case of subdomain takeover.

![image](https://github.com/EdOverflow/can-i-take-over-xyz/assets/46958133/518f7014-52ba-4e63-98ab-bde98ce5c96c)

**Comment by M359AH** [comment]
Yes I think the exploitation now will not complete

**Comment by rew1nter** [comment]
Shouldn't this be marked not vulnerable at this point?

**Comment by jan-muhammad-zaidi** [comment]
It should be closed as Not Vulnerable

**Comment by Atharv34** [comment]
Edge Case.
![Screenshot 2024-04-18 230749](https://github.com/EdOverflow/can-i-take-over-xyz/assets/55253073/21d2164c-a7fb-4506-a48a-360fabe9e88b)

**Comment by Atharv34** [comment]
Edge Case.
![Screenshot 2024-04-18 231240](https://github.com/EdOverflow/can-i-take-over-xyz/assets/55253073/3abfd135-2fe4-40dd-b0e9-5326a1124cf7)

**Comment by pdelteil** [comment]
> Edge Case. ![Screenshot 2024-04-18 231240](https://private-user-images.githubusercontent.com/55253073/323706832-3abfd135-2fe4-40dd-b0e9-5326a1124cf7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ1MDYzOTUsIm5iZiI6MTcxNDUwNjA5NSwicGF0aCI6Ii81NTI1MzA3My8zMjM3MDY4MzItM2FiZmQxMzUtMmZlNC00MGRkLWIwZTktNTMyNmExMTI0Y2Y3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDMwVDE5NDEzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlY2RjYTkyNGYyYzFkZTk0MDc3YTkyMTIxNmYwOTZlNGI0MTA0N2U5ZDE1ZTg3NTMzMzg4OGVhMDgxZGIwMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Bg0lIQDHYcE7_ZC-LZ8xGfXFUlYQJYxnSZ35jsR4fcI)

This is not the example of an edge case. Edge case would be if you managed to take over the subdomain due to uncommon or unknown conditions.

**Comment by zangcc** [comment]
Is this vulnerability no longer exploitable? Why hasn't the Status changed to Not vulnerable?
<img width="944" alt="image" src="https://github.com/EdOverflow/can-i-take-over-xyz/assets/64825932/54beb2bf-1769-4ef1-8073-ba8f64e741ca">

**Comment by pradyumna14** [comment]
lol, i was happy to find a 404 and tried to put that subdomain for my project index.html, but then i got the shock as it required ownership. 

Did someone bypass this though?

**Comment by saad-cpu-ops** [comment]
damn also me> I discovered a subdomain takeover (STO) vulnerability across 7 subdomains. Each has a CNAME record pointing to Vercel (cname.vercel-dns.com), but the underlying Vercel deployment no longer exists — confirmed by requesting each host and consistently receiving Vercel's DEPLOYMENT_NOT_FOUND response, re-verified live on the date of the report. I attempted to complete the takeover by adding each hostname to a new project in my own Vercel account. Vercel detected that these hostnames were previously registered under a different account and required a TXT record be added to the domain's real DNS :) anyone bypassed this
